### PR TITLE
[Utility] Added Random to tiny tank & New obstacle Mud

### PR DESCRIPTION
**Pull request by teddyxwx**

### CODE REVIEW

**Comment by teddyxwx** [comment]
加入了随机因素，tank在移动时会产生少量偏移，发出的炮弹也会产生少量偏移

File: src/battle_game/core/units/tiny_tank.cpp [hunk 1]
```diff
@@ -86,25 +89,42 @@ void Tank::Update() {
...
     auto new_position =
         position_ + glm::vec2{glm::rotate(glm::mat4{1.0f}, rotation_,
                                           glm::vec3{0.0f, 0.0f, 1.0f}) *
                               glm::vec4{offset, 0.0f, 0.0f}};
+    // printf("Before:%f %f\n",new_position.x,new_position.y);
+    if (fx) {
+      glm::vec2 offset2{0.0f};
+      // printf("%d\n",rand());
+      offset2.y += (rand() - (1 << 30)) / 1000000000.0;
+      offset2 *= kSecondPerTick * speed;
```
Comment (LazyJazz) [inline src/battle_game/core/units/tiny_tank.cpp:117]: Why are you using rand() function? I gave you RandomFloat() in GameCore and told you in docs. Undo this!
BTW, **DO NOT** make changes in the example code.

File: src/battle_game/core/units/tiny_tank_xwx.cpp [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  auto player = game_core_->GetPlayer(player_id_);
+  if (player) {
+    auto &input_data = player->GetInputData();
+    auto diff = input_data.mouse_cursor_position - position_;
+    if (glm::length(diff) < 1e-4) {
+      turret_rotation_ = rotation_;
+    }
+    turret_rotation_ = std::atan2(diff.y, diff.x) - glm::radians(90.0f);
+  }
+}
```
Comment (LazyJazz) [inline src/battle_game/core/units/tiny_tank_xwx.cpp:126]: Inherit from unit::Tank if you are reusing the code.

**Review by LazyJazz — CHANGES_REQUESTED** [review]

**Comment by teddyxwx** [comment]
加入了mud，不会对子弹产生影响，但是会让经过的坦克速度大幅下降

File: src/battle_game/core/units/random_tank.cpp [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    if (input_data.key_down[GLFW_KEY_A]) {
+      rotation_offset += 1.0f;
+    }
+    if (input_data.key_down[GLFW_KEY_D]) {
+      rotation_offset -= 1.0f;
+    }
+    rotation_offset *= kSecondPerTick * rotate_angular_speed * GetSpeedScale();
+    game_core_->PushEventRotateUnit(id_, rotation_ + rotation_offset);
+  }
+}
```
Comment (LazyJazz) [inline src/battle_game/core/units/random_tank.cpp:88]: Did you make any change? if not, inherit from unit::Tank too.

Comment (teddyxwx) [reply src/battle_game/core/units/random_tank.cpp:88]: I added the speed reduction of mud and random move in this function

**Pull request closed without merging**